templates/CRM/Group/Form/Search.tpl

```diff
@@ -99,12 +99,13 @@
         "url": {/literal}'{crmURL p="civicrm/ajax/grouplist" h=0 q="snippet=4"}'{literal},
         "data": function (d) {
 
-          var groupTypes = ($('.crm-group-search-form-block #group_type_search_1').prop('checked')) ? '1' : '';
-          if (groupTypes) {
-            groupTypes = ($('.crm-group-search-form-block #group_type_search_2').prop('checked')) ? groupTypes + ',2' : groupTypes;
-          } else {
-            groupTypes = ($('.crm-group-search-form-block #group_type_search_2').prop('checked')) ? '2' : '';
-          }
+          var groupTypes = [];
+          $('input[id*="group_type_search_"]').each(function(e) {
+            if ($(this).prop('checked')) {
+              groupTypes.push($(this).attr('id').replace(/group_type_search_/, ''));
+            }
+          });
+          groupTypes = groupTypes.join();
```

Comment:
I don't like that `var groupTypes` changes from array to string; it makes code harder to read for future maintainers. I suggest using two separate vars, one array, and one the joined string.